pkg/k8s/client.go

```diff
@@ -463,3 +484,150 @@ func (c *Client) GetMeshConfig() configv1alpha2.MeshConfig {
 func (c *Client) GetOSMNamespace() string {
 	return c.osmNamespace
 }
+
+// ListEgressPoliciesForSourceIdentity lists the Egress policies for the given source identity based on service accounts
+func (c *Client) ListEgressPoliciesForSourceIdentity(source identity.K8sServiceAccount) []*policyv1alpha1.Egress {
```

Comment:
Egress policy, IngressBackend etc. are OSM/SMI terminologies. I think they are not part of compute.Interface as they are not Kubernetes native.

I will update the method names

Reply:
For the 3 `GetUpstreamTrafficSettingBy<Type>()` methods, I don't find a better way to name them. They are split from the original `GetUpstreamTrafficSetting()` method. Do you have any suggestion?

Reply:
They are being queried from Kubernetes though. The interface is to the control plane of where they are housed. 

So it would looks like this:

in compute/kube/client.go, are all the methods that are defined here.

In k8s/client.go, you have: `ListEgressPolicies`, which just calls list on the informer.

Right now there's logic in k8s, which makes k8s aware of OSM primitives and mappings. The k8s package will have simple CRUD based methods (get, delete, update, create, list), and nothing more. Happy to sync if you'd like